test/integration/acceptance/add_peer_test.cpp

```diff
@@ -0,0 +1,325 @@
+
+    auto permissions =
+        interface::RolePermissionSet({Role::kReceive, Role::kTransfer});
+
+    return itf_
+        ->sendTxAwait(makeUserWithPerms(permissions), checkBlockHasNTxs<1>)
+        .sendTxAwait(
+            complete(baseTx(kAdminId).addAssetQuantity(kAssetId, "20000.0"),
+                     kAdminKeypair),
+            checkBlockHasNTxs<1>);
+  }
+
+ protected:
+  void SetUp() override {
+    itf_ =
+        std::make_unique<IntegrationTestFramework>(1, boost::none, true, true);
+    itf_->initPipeline(kAdminKeypair);
+  }
+
+  std::vector<std::shared_ptr<FakePeer>> fake_peers_;
+};
+
+auto makePeerPointeeMatcher(interface::types::AddressType address,
+                            interface::types::PubkeyType pubkey) {
+  return ::testing::Truly(
+      [address = std::move(address),
+       pubkey = std::move(pubkey)](std::shared_ptr<interface::Peer> peer) {
+        return peer->address() == address and peer->pubkey() == pubkey;
+      });
+}
+
+auto makePeerPointeeMatcher(std::shared_ptr<interface::Peer> peer) {
+  return makePeerPointeeMatcher(peer->address(), peer->pubkey());
+}
+
+/**
+ * @given a network of single peer
+ * @when it receives a valid signed addPeer command
+ * @then the transaction is committed
+ *    @and the WSV reports that there are two peers: the initial and the added one
+ */
+TEST_F(FakePeerExampleFixture, FakePeerIsAdded) {
+  // init the real peer with no other peers in the genesis block
+  auto &itf = prepareState();
+
+  const std::string new_peer_address = "127.0.0.1:1234";
+  const auto new_peer_pubkey =
+      shared_model::crypto::DefaultCryptoAlgorithmType::generateKeypair()
+          .publicKey();
+
+  itf.sendTxAwait(
+      complete(baseTx(kAdminId).addPeer(new_peer_address, new_peer_pubkey),
+               kAdminKeypair),
+      checkBlockHasNTxs<1>);
+
+  auto opt_peers = itf.getIrohaInstance()
+                       .getIrohaInstance()
+                       ->getStorage()
+                       ->createPeerQuery()
+                       .value()
+                       ->getLedgerPeers();
+
+  ASSERT_TRUE(opt_peers);
+  EXPECT_THAT(*opt_peers,
+              ::testing::UnorderedElementsAre(
+                  makePeerPointeeMatcher(itf.getThisPeer()),
+                  makePeerPointeeMatcher(new_peer_address, new_peer_pubkey)));
+}
+
+/**
+ * @given a network of single peer
+ * @when it receives a not fully signed transaction and then a new peer is added
+ * @then the first peer propagates MST state to the newly added peer
+ */
+TEST_F(FakePeerExampleFixture, MstStatePropagtesToNewPeer) {
+  // init the real peer with no other peers in the genesis block
+  auto &itf = prepareState();
+
+  // then create a fake peer
+  auto new_peer = itf.addFakePeer(boost::none);
+  auto mst_states_observable = new_peer->getMstStatesObservable().replay();
+  mst_states_observable.connect();
+  auto new_peer_server = new_peer->run();
+
+  // and add it with addPeer
+  itf.sendTxAwait(
```

Comment:
Why we don't wait for the commit of the transaction with addPeer?

Reply:
`sendTxAwait` waits for the commit:
https://github.com/hyperledger/iroha/blob/98e4eebbafc0ebef10f7f3ad6f9bcdf61cdd8bd1/test/framework/integration_framework/integration_test_framework.cpp#L497-L502